### PR TITLE
Add TryClone trait

### PR DESCRIPTION
**Pull request by ava57r**
Drop macro `impl_native_index_clone`

This is for generic code

### CODE REVIEW

**Comment by ava57r** [comment]
@Enet4 Could You review it?

**Review by Enet4 — COMMENTED** [review]
It was nice that we got rid of the macro and replaced it with something that can be applied to all implementations! :+1:
It is a breaking change, but it's a reasonable one to introduce.

I found no problems in the code. Just suggesting a minor documentation tweak before we merge.

File: src/index/mod.rs [hunk 1]
```diff
@@ -322,6 +322,26 @@ pub trait TryFromInnerPtr: NativeIndex {
         Self: Sized;
 }
 
+/// A macro which provides a Clone implementation to native index types.
+pub trait TryClone: FromInnerPtr {
```
[the remark this answers is not in the document]
Comment (ava57r) [reply src/index/mod.rs:326]: I want to refresh my mind about PR.
I added trait `TryFromInnerPtr` early. It is `safe` interface.

**Review by Enet4 — APPROVED** [review]
Thanks!

**Pull request merged**